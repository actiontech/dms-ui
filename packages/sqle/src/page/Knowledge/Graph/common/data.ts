import { SigmaContainerProps } from '@react-sigma/core';
import { EdgeType, NodeType } from '../index.type';
import { Attributes } from 'graphology-types';

export const sigmaSettings: SigmaContainerProps<
  NodeType,
  EdgeType,
  Attributes
>['settings'] = {
  // 允许在无效容器中渲染 (默认值: false)
  allowInvalidContainer: true,
  defaultNodeType: 'circle',
  // 设置边的默认类型 (默认值: 'line')
  defaultEdgeType: 'arrow',

  // 相机缩放比例限制 (默认值: min: 0.1, max: 32)
  minCameraRatio: 0.2,
  maxCameraRatio: 5,

  // 节点渲染程序类 (默认值: {})
  nodeProgramClasses: {},

  // 渲染性能优化
  hideEdgesOnMove: true, // 移动时隐藏边 (默认值: false)
  hideLabelsOnMove: false, // 移动时隐藏标签 (默认值: false)
  renderLabels: true, // 是否渲染标签 (默认值: true)
  renderEdgeLabels: false, // 是否渲染边标签 (默认值: false)

  // 标签渲染优化
  labelDensity: 0.1, // 标签密度 (默认值: 0.03)
  labelGridCellSize: 100, // 标签网格大小 (默认值: 100)
  labelRenderedSizeThreshold: 8, // 标签大小阈值 (默认值: 6)

  // 默认样式
  defaultNodeColor: '#1a73e8', // 节点颜色 (默认值: '#999')
  defaultEdgeColor: '#ccc', // 边颜色 (默认值: '#ccc')

  // 交互体验优化
  zoomToSizeRatioFunction: (x: number) => x // 缩放比例函数 (默认值: Math.pow)
};
