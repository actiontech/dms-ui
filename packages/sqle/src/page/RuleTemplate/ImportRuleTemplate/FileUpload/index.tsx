import { Button, Form, Upload, UploadProps } from 'antd';
import { BasicButton, EmptyBox, ReminderInformation } from '@actiontech/shared';
import { useTranslation } from 'react-i18next';
import { useBoolean } from 'ahooks';
import {
  FileUploadCheckStatusType,
  SelectFileFormFields
} from '../../../../hooks/useRuleTemplateForm/index.type';
import { getRuleTemplateFileV1FileTypeEnum } from '@actiontech/shared/lib/api/sqle/service/rule_template/index.enum';
import DownloadTemplateModal from './components/DownloadTemplateModal';

type Props = UploadProps & {
  uploadCheckStatus: FileUploadCheckStatusType;
};

const FileUpload: React.FC<Props> = ({ uploadCheckStatus, ...props }) => {
  const { t } = useTranslation();

  const form = Form.useFormInstance<SelectFileFormFields>();

  const fileType = Form.useWatch('fileType', form);

  const [
    downloadTemplateModalVisibility,
    { setTrue: openModal, setFalse: closeModal }
  ] = useBoolean();

  return (
    <div className="vertical-flex">
      <div className="flex-display">
        <Upload {...props}>
          <BasicButton>{t('common.upload')}</BasicButton>
        </Upload>
        <Button type="link" onClick={openModal}>
          {t('ruleTemplate.importRuleTemplate.downloadTemplate')}
        </Button>
      </div>

      <EmptyBox if={!!uploadCheckStatus.success}>
        <ReminderInformation
          status="success"
          message={t('ruleTemplate.importRuleTemplate.checkSuccess')}
        />
      </EmptyBox>
      <EmptyBox if={!!uploadCheckStatus.errorMessage}>
        <ReminderInformation
          status="error"
          message={uploadCheckStatus.errorMessage ?? ''}
        />
      </EmptyBox>

      <DownloadTemplateModal
        fileType={fileType as unknown as getRuleTemplateFileV1FileTypeEnum}
        open={downloadTemplateModalVisibility}
        onClose={closeModal}
      />
    </div>
  );
};

export default FileUpload;
