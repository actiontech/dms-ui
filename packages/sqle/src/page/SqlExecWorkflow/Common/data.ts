export const SOURCE_WORKFLOW_PATH_KEY = 'sourceWorkflowId';
