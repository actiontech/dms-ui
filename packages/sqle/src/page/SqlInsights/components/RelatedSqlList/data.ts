import { GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum } from '@actiontech/shared/lib/api/sqle/service/SqlInsight/index.enum';
import { t } from '../../../../locale';

export const RelatedSqlFilterSourceOptions = [
  {
    label: t('sqlInsights.relatedSqlList.source.order'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.workflow
  },
  {
    label: t('sqlInsights.relatedSqlList.source.sqlManage'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.sql_manage
  },
  {
    label: t('sqlInsights.relatedSqlList.source.workbench'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.workbench
  }
];
