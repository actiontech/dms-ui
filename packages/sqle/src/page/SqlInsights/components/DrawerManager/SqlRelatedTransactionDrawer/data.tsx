import { TransactionInfoLockTypeEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';
import { t } from '../../../../../locale';

export const TransactionInfoLockTypeEnumDict = {
  [TransactionInfoLockTypeEnum.EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.exclusive'
  ),
  [TransactionInfoLockTypeEnum.SHARED]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.shared'
  ),
  [TransactionInfoLockTypeEnum.INTENTION_SHARED]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.intentionShared'
  ),
  [TransactionInfoLockTypeEnum.INTENTION_EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.intentionExclusive'
  ),
  [TransactionInfoLockTypeEnum.SHARED_INTENTION_EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.sharedIntentionExclusive'
  ),
  [TransactionInfoLockTypeEnum.ROW_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.rowLock'
  ),
  [TransactionInfoLockTypeEnum.TABLE_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.tableLock'
  ),
  [TransactionInfoLockTypeEnum.METADATA_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.metadataLock'
  )
};

export const TransactionInfoLockTypeEnumFlagDict = {
  [TransactionInfoLockTypeEnum.SHARED]: 'S',
  [TransactionInfoLockTypeEnum.INTENTION_SHARED]: 'IS',
  [TransactionInfoLockTypeEnum.INTENTION_EXCLUSIVE]: 'IX',
  [TransactionInfoLockTypeEnum.SHARED_INTENTION_EXCLUSIVE]: 'SIX',
  [TransactionInfoLockTypeEnum.ROW_LOCK]: 'RL',
  [TransactionInfoLockTypeEnum.TABLE_LOCK]: 'TL',
  [TransactionInfoLockTypeEnum.METADATA_LOCK]: 'ML',
  [TransactionInfoLockTypeEnum.EXCLUSIVE]: 'X'
};
