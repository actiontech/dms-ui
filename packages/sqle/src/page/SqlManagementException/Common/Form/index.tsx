import { Alert, Form, Radio } from 'antd';
import { useTranslation } from 'react-i18next';
import { SqlManagementExceptionFormProps } from '../../index.type';
import { whiteSpaceSql } from '@actiontech/shared/lib/utils/FormRule';
import {
  useMonacoEditor,
  MonacoEditor
} from '@actiontech/shared/lib/components/MonacoEditor';
import { CreateBlacklistReqV1TypeEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';
import { DrawerFormLayout } from '@actiontech/shared/lib/data/common';
import { BasicInput, EmptyBox, BasicSelect } from '@actiontech/shared';
import { SqlManagementExceptionMatchTypeOptions } from '../../index.data';
import useInstance from '../../../../hooks/useInstance';
import { useEffect } from 'react';
import { useCurrentProject } from '@actiontech/shared/lib/features';
import { WarningFilled } from '@actiontech/icons';

const SqlManagementExceptionForm: React.FC<SqlManagementExceptionFormProps> = ({
  form,
  isUpdate
}) => {
  const { t } = useTranslation();

  const { editorDidMount } = useMonacoEditor(form, { formName: 'sql' });

  const matchType = Form.useWatch('type', form);

  const { updateInstanceList, generateInstanceSelectOption, loading } =
    useInstance();

  const { projectName } = useCurrentProject();

  useEffect(() => {
    updateInstanceList({ project_name: projectName });
  }, [updateInstanceList, projectName]);

  return (
    <Form form={form} layout="vertical" {...DrawerFormLayout}>
      <Form.Item
        label={t('sqlManagementException.table.matchType')}
        name="type"
        initialValue={CreateBlacklistReqV1TypeEnum.sql}
      >
        <Radio.Group
          options={SqlManagementExceptionMatchTypeOptions}
          onChange={() => form.resetFields(['content'])}
        />
      </Form.Item>
      <Form.Item label={t('sqlManagementException.table.desc')} name="desc">
        <BasicInput.TextArea
          className="textarea-no-resize"
          autoSize={{
            minRows: 3,
            maxRows: 10
          }}
          placeholder={t('common.form.placeholder.input')}
        />
      </Form.Item>
      <EmptyBox
        if={
          matchType === CreateBlacklistReqV1TypeEnum.sql ||
          matchType === CreateBlacklistReqV1TypeEnum.fp_sql
        }
      >
        <Form.Item
          name="sql"
          label={t('sqlManagementException.modal.sql')}
          rules={[
            {
              required: true
            },
            ...whiteSpaceSql()
          ]}
          initialValue="/* input your sql */"
        >
          <MonacoEditor
            width="100%"
            height="500px"
            language="sql"
            onMount={editorDidMount}
          />
        </Form.Item>
      </EmptyBox>
      <EmptyBox if={matchType === CreateBlacklistReqV1TypeEnum.instance}>
        <Form.Item
          name="instance"
          label={t('sqlManagementException.matchType.instance')}
          rules={[
            {
              required: true
            }
          ]}
        >
          <BasicSelect loading={loading}>
            {generateInstanceSelectOption()}
          </BasicSelect>
        </Form.Item>
      </EmptyBox>
      <EmptyBox
        if={
          matchType === CreateBlacklistReqV1TypeEnum.ip ||
          matchType === CreateBlacklistReqV1TypeEnum.cidr ||
          matchType === CreateBlacklistReqV1TypeEnum.host
        }
      >
        <Form.Item
          name={matchType}
          label={t(`sqlManagementException.matchType.${matchType}`)}
          rules={[
            {
              required: true
            }
          ]}
        >
          <BasicInput />
        </Form.Item>
      </EmptyBox>

      <EmptyBox if={isUpdate}>
        <Alert
          showIcon
          icon={<WarningFilled />}
          message={t('sqlManagementException.modal.update.tips')}
          type="warning"
        />
      </EmptyBox>
    </Form>
  );
};

export default SqlManagementExceptionForm;
