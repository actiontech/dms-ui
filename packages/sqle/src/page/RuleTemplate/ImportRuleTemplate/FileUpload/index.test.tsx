import { act, fireEvent, renderHook, screen } from '@testing-library/react';
import FileUpload from '.';
import { sqleSuperRender } from '../../../../testUtils/superRender';
import { useForm } from 'antd/es/form/Form';
import { Form, UploadProps } from 'antd';
import { FileUploadCheckStatusType } from '../../../../hooks/useRuleTemplateForm/index.type';
import { mockDatabaseType } from '../../../../testUtils/mockHooks/mockDatabaseType';
import rule_template from '../../../../testUtils/mockApi/rule_template';
import { mockUseDatabaseType } from '../../../../testUtils/mockHooks/data';
import { selectOptionByIndex } from '@actiontech/shared/lib/testUtil/customQuery';

describe('test FileUpload', () => {
  const customRender = (
    props: UploadProps & {
      uploadCheckStatus: FileUploadCheckStatusType;
    }
  ) => {
    const { result } = renderHook(() => useForm());
    return sqleSuperRender(
      <Form form={result.current[0]}>
        <Form.Item name="fileType" initialValue="csv">
          <input />
        </Form.Item>
        <FileUpload {...props} />
      </Form>
    );
  };
  let getRuleTemplateFileSpy: jest.SpyInstance;
  beforeEach(() => {
    jest.useFakeTimers();
    getRuleTemplateFileSpy = rule_template.getRuleTemplateFile();
    mockDatabaseType({ dbTypeOptions: [{ label: 'MySQL', value: 'MySQL' }] });
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  it('should display a success message when uploadCheckStatus.success is truthy', () => {
    const { baseElement } = customRender({
      uploadCheckStatus: { success: true }
    });

    expect(baseElement).toMatchSnapshot();
  });

  it('should display an error message when uploadCheckStatus.errorMessage is present', () => {
    const { baseElement } = customRender({
      uploadCheckStatus: { success: false, errorMessage: 'error message' }
    });

    expect(baseElement).toMatchSnapshot();
  });

  it('should render the DownloadTemplateModal component when downloadTemplateModalVisibility is true', async () => {
    const { baseElement } = customRender({
      uploadCheckStatus: { success: false, errorMessage: 'error message' }
    });

    fireEvent.click(screen.getByText('下载导入模板'));

    expect(mockUseDatabaseType.updateDriverNameList).toHaveBeenCalledTimes(1);

    await act(async () => jest.advanceTimersByTime(3000));

    expect(baseElement).toMatchSnapshot();

    selectOptionByIndex('数据源类型', 'MySQL');

    await act(async () => jest.advanceTimersByTime(0));

    fireEvent.click(screen.getByText('下 载'));
    await act(async () => jest.advanceTimersByTime(0));

    expect(getRuleTemplateFileSpy).toHaveBeenCalledTimes(1);
    expect(getRuleTemplateFileSpy).toHaveBeenCalledWith(
      { instance_type: 'MySQL', file_type: 'csv' },
      { responseType: 'blob' }
    );
    expect(screen.getByText('正在下载模板文件...')).toBeInTheDocument();

    await act(async () => jest.advanceTimersByTime(3000));
    expect(screen.queryByText('正在下载模板文件...')).not.toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
