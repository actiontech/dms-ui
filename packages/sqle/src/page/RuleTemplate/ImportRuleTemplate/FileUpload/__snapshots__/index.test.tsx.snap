// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test FileUpload should display a success message when uploadCheckStatus.success is truthy 1`] = `
<body>
  <div />
  <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  id="fileType"
                  value="csv"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="vertical-flex"
      >
        <div
          class="flex-display"
        >
          <span
            class="ant-upload-wrapper"
          >
            <div
              class="ant-upload ant-upload-select"
            >
              <span
                class="ant-upload"
                role="button"
                tabindex="0"
              >
                <input
                  accept=""
                  style="display: none;"
                  type="file"
                />
                <button
                  class="ant-btn ant-btn-default basic-button-wrapper css-1uxy5k6"
                  type="button"
                >
                  <span>
                    选择文件上传
                  </span>
                </button>
              </span>
            </div>
            <div
              class="ant-upload-list ant-upload-list-text"
            />
          </span>
          <button
            class="ant-btn ant-btn-link"
            type="button"
          >
            <span>
              下载导入模板
            </span>
          </button>
        </div>
        <div
          class="ant-space ant-space-horizontal ant-space-align-start css-1nec0wd"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <span
              class="css-u3jcp4"
            >
              <svg
                class="custom-icon"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 14.667A6.667 6.667 0 1 1 8 1.333a6.667 6.667 0 0 1 0 13.334m0-1.334A5.333 5.333 0 1 0 8 2.667a5.333 5.333 0 0 0 0 10.666m-.665-2.666L4.507 7.838l.942-.943 1.886 1.886 3.771-3.771.943.943z"
                  fill="#1CB889"
                />
              </svg>
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              class="ant-typography ant-typography-success"
            >
              <pre>
                校验通过
              </pre>
            </span>
          </div>
        </div>
      </div>
    </form>
  </div>
</body>
`;

exports[`test FileUpload should display an error message when uploadCheckStatus.errorMessage is present 1`] = `
<body>
  <div />
  <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  id="fileType"
                  value="csv"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="vertical-flex"
      >
        <div
          class="flex-display"
        >
          <span
            class="ant-upload-wrapper"
          >
            <div
              class="ant-upload ant-upload-select"
            >
              <span
                class="ant-upload"
                role="button"
                tabindex="0"
              >
                <input
                  accept=""
                  style="display: none;"
                  type="file"
                />
                <button
                  class="ant-btn ant-btn-default basic-button-wrapper css-1uxy5k6"
                  type="button"
                >
                  <span>
                    选择文件上传
                  </span>
                </button>
              </span>
            </div>
            <div
              class="ant-upload-list ant-upload-list-text"
            />
          </span>
          <button
            class="ant-btn ant-btn-link"
            type="button"
          >
            <span>
              下载导入模板
            </span>
          </button>
        </div>
        <div
          class="ant-space ant-space-horizontal ant-space-align-start css-1nec0wd"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <span
              class="css-u3jcp4"
            >
              <svg
                class="custom-icon"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 14.667A6.666 6.666 0 1 1 8 1.334a6.666 6.666 0 0 1 0 13.333m0-1.334A5.333 5.333 0 1 0 8 2.667a5.333 5.333 0 0 0 0 10.666m0-6.276 1.885-1.886.944.944L8.943 8l1.886 1.885-.944.944L8 8.943l-1.885 1.886-.944-.944L7.057 8 5.171 6.115l.944-.944z"
                  fill="#F66074"
                />
              </svg>
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              class="ant-typography ant-typography-danger"
            >
              <pre>
                error message
              </pre>
            </span>
          </div>
        </div>
      </div>
    </form>
  </div>
</body>
`;

exports[`test FileUpload should render the DownloadTemplateModal component when downloadTemplateModalVisibility is true 1`] = `
<body>
  <div />
  <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  id="fileType"
                  value="csv"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="vertical-flex"
      >
        <div
          class="flex-display"
        >
          <span
            class="ant-upload-wrapper"
          >
            <div
              class="ant-upload ant-upload-select"
            >
              <span
                class="ant-upload"
                role="button"
                tabindex="0"
              >
                <input
                  accept=""
                  style="display: none;"
                  type="file"
                />
                <button
                  class="ant-btn ant-btn-default basic-button-wrapper css-1uxy5k6"
                  type="button"
                >
                  <span>
                    选择文件上传
                  </span>
                </button>
              </span>
            </div>
            <div
              class="ant-upload-list ant-upload-list-text"
            />
          </span>
          <button
            class="ant-btn ant-btn-link"
            type="button"
          >
            <span>
              下载导入模板
            </span>
          </button>
        </div>
        <div
          class="ant-space ant-space-horizontal ant-space-align-start css-1nec0wd"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <span
              class="css-u3jcp4"
            >
              <svg
                class="custom-icon"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 14.667A6.666 6.666 0 1 1 8 1.334a6.666 6.666 0 0 1 0 13.333m0-1.334A5.333 5.333 0 1 0 8 2.667a5.333 5.333 0 0 0 0 10.666m0-6.276 1.885-1.886.944.944L8.943 8l1.886 1.885-.944.944L8 8.943l-1.885 1.886-.944-.944L7.057 8 5.171 6.115l.944-.944z"
                  fill="#F66074"
                />
              </svg>
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              class="ant-typography ant-typography-danger"
            >
              <pre>
                error message
              </pre>
            </span>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal basic-modal-wrapper css-vvpsbb"
          role="dialog"
          style="width: 480px; transform-origin: NaNpx NaNpx;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <button
              aria-label="Close"
              class="ant-modal-close"
              type="button"
            >
              <span
                class="ant-modal-close-x"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 14 14"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m7 6.175 2.887-2.887.825.825L7.825 7l2.887 2.888-.825.824L7 7.825l-2.888 2.887-.824-.824L6.175 7 3.288 4.113l.824-.825z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                下载导入模板
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal"
              >
                <div
                  class="ant-form-item has-required-style css-1jlm9cy"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class="ant-form-item-required"
                        for="dbType"
                        title="数据源类型"
                      >
                        数据源类型
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            aria-required="true"
                            class="ant-select ant-select-in-form-item basic-select-wrapper css-iy4o4x ant-select-single ant-select-show-arrow"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-activedescendant="dbType_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="dbType_list"
                                  aria-expanded="false"
                                  aria-haspopup="listbox"
                                  aria-owns="dbType_list"
                                  aria-required="true"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="dbType"
                                  readonly=""
                                  role="combobox"
                                  style="opacity: 0;"
                                  type="search"
                                  unselectable="on"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              >
                                请选择{{name}}
                              </span>
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn ant-btn-default basic-button-wrapper css-1uxy5k6"
                type="button"
              >
                <span>
                  取 消
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary basic-button-wrapper css-1uxy5k6"
                type="button"
              >
                <span>
                  下 载
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`test FileUpload should render the DownloadTemplateModal component when downloadTemplateModalVisibility is true 2`] = `
<body>
  <div />
  <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  id="fileType"
                  value="csv"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="vertical-flex"
      >
        <div
          class="flex-display"
        >
          <span
            class="ant-upload-wrapper"
          >
            <div
              class="ant-upload ant-upload-select"
            >
              <span
                class="ant-upload"
                role="button"
                tabindex="0"
              >
                <input
                  accept=""
                  style="display: none;"
                  type="file"
                />
                <button
                  class="ant-btn ant-btn-default basic-button-wrapper css-1uxy5k6"
                  type="button"
                >
                  <span>
                    选择文件上传
                  </span>
                </button>
              </span>
            </div>
            <div
              class="ant-upload-list ant-upload-list-text"
            />
          </span>
          <button
            class="ant-btn ant-btn-link"
            type="button"
          >
            <span>
              下载导入模板
            </span>
          </button>
        </div>
        <div
          class="ant-space ant-space-horizontal ant-space-align-start css-1nec0wd"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <span
              class="css-u3jcp4"
            >
              <svg
                class="custom-icon"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 14.667A6.666 6.666 0 1 1 8 1.334a6.666 6.666 0 0 1 0 13.333m0-1.334A5.333 5.333 0 1 0 8 2.667a5.333 5.333 0 0 0 0 10.666m0-6.276 1.885-1.886.944.944L8.943 8l1.886 1.885-.944.944L8 8.943l-1.885 1.886-.944-.944L7.057 8 5.171 6.115l.944-.944z"
                  fill="#F66074"
                />
              </svg>
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              class="ant-typography ant-typography-danger"
            >
              <pre>
                error message
              </pre>
            </span>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-wrap"
        style="display: none;"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal basic-modal-wrapper css-vvpsbb"
          role="dialog"
          style="width: 480px; transform-origin: NaNpx NaNpx; display: none;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <button
              aria-label="Close"
              class="ant-modal-close"
              type="button"
            >
              <span
                class="ant-modal-close-x"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 14 14"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m7 6.175 2.887-2.887.825.825L7.825 7l2.887 2.888-.825.824L7 7.825l-2.888 2.887-.824-.824L6.175 7 3.288 4.113l.824-.825z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                下载导入模板
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal"
              >
                <div
                  class="ant-form-item has-required-style css-1jlm9cy"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class="ant-form-item-required"
                        for="dbType"
                        title="数据源类型"
                      >
                        数据源类型
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            aria-required="true"
                            class="ant-select ant-select-in-form-item basic-select-wrapper css-iy4o4x ant-select-single ant-select-show-arrow"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-activedescendant="dbType_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="dbType_list"
                                  aria-expanded="false"
                                  aria-haspopup="listbox"
                                  aria-owns="dbType_list"
                                  aria-required="true"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="dbType"
                                  readonly=""
                                  role="combobox"
                                  style="opacity: 0;"
                                  type="search"
                                  unselectable="on"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              >
                                请选择{{name}}
                              </span>
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn ant-btn-default basic-button-wrapper css-1uxy5k6"
                disabled=""
                type="button"
              >
                <span>
                  取 消
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary basic-button-wrapper css-1uxy5k6"
                disabled=""
                type="button"
              >
                <span>
                  下 载
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
